ui/components/app/flask/snap-authorship/snap-authorship.js

```diff
@@ -43,48 +54,129 @@ const SnapAuthorship = ({ snapId, className }) => {
 
   const friendlyName = snapId && getSnapName(snapId, subjectMetadata);
 
+  if (expanded && !snap) {
```

Comment:
I wonder if there is a state where `snap` is undefined and this would throw. Similarly to how `snapId` can be undefined?